infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -380,7 +395,9 @@ private LogData getLogData(LogEntry entry) {
      * @throws IOException IO exception
      */
     private Metadata parseMetadata(FileChannel fileChannel) throws IOException {
-        if (fileChannel.size() - fileChannel.position() < METADATA_SIZE) {
+        long actualMetaDataSize = fileChannel.size() - fileChannel.position();
+        if (actualMetaDataSize < METADATA_SIZE) {
+            log.debug("Meta data has wrong size. Actual size: {}, expected: {}", actualMetaDataSize, METADATA_SIZE);
             return null;
```

Comment:
Do you not want to short-circuit in this case?

Reply:
yep, I want to make it more explicit, in the prev version it was a bit unclear what means `fileChannel.size() - fileChannel.position();`

Reply:
Previously a null value was being returned. Now, we just continue on... Why are we continuing when we know the metadata is corrupt?

Reply:
In `parseHeader()` we check if metadata is null. It's a bit different behaviour when it's null or when we throw an exception. Right now I am not ready to fix/simplify that logic. In next PR-s I will do it